exporter/exporter.go

```diff
@@ -221,6 +229,58 @@ func (ne *NATSExporter) generateTLSConfig() (*tls.Config, error) {
 	return config, nil
 }
 
+// isBcrypt checks whether the given password or token is bcrypted.
+func isBcrypt(password string) bool {
+	return strings.HasPrefix(password, bcryptPrefix)
+}
+
+func (ne *NATSExporter) isValidUserPass(user, password string) bool {
+	if user != ne.opts.HTTPUser {
+		return false
+	}
+	exporterPassword := ne.opts.HTTPPassword
+	if isBcrypt(exporterPassword) {
+		if err := bcrypt.CompareHashAndPassword([]byte(exporterPassword), []byte(password)); err != nil {
+			return false
+		}
+	} else if exporterPassword != password {
+		return false
+	}
+	return true
+}
+
+// getScrapeHandler returns the defualt handler if no nttp
+// auhtorization has been specificed.  Otherwise, it checks
+// basic authorization.
+func (ne *NATSExporter) getScrapeHandler() http.Handler {
+	h := promhttp.Handler()
+
+	if ne.opts.HTTPUser != "" {
+		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			auth := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
+			if len(auth) != 2 || auth[0] != "Basic" {
+				http.Error(rw, "authorization failed", http.StatusUnauthorized)
+				return
+			}
+
+			payload, err := base64.StdEncoding.DecodeString(auth[1])
+			if err != nil {
+				http.Error(rw, "authorization failed", http.StatusBadRequest)
```

Comment:
missing `return` here I think?

Reply:
Good catch - Thanks!

Reply:
I wouldn't do BadRequest on failing to decode the authorization. I would simply fail it with the Unauthorized